File: app/src/main/java/workshop/akbolatss/tagsnews/model/RssSourceRepository.java
package workshop.akbolatss.tagsnews.model;

import java.util.List;

import io.reactivex.Single;
import workshop.akbolatss.tagsnews.model.dao.RssSource;


public interface RssSourceRepository {

    Single<List<RssSource>> getAllSources();

    Single<List<RssSource>> getOnlyActive();

    Single<FeedlyResponse> getQueryResult(String query);

    void initDefaultSource(RssSource source);

    void addNewSource(RssSource source);

    void updateSource(RssSource source);

    void swapSources(RssSource from, RssSource to);

    void deleteSource(RssSource source);
}

File: app/src/main/java/workshop/akbolatss/tagsnews/model/RssSourceRepository.kt
package workshop.akbolatss.tagsnews.model

import io.reactivex.Single
import workshop.akbolatss.tagsnews.model.dao.RssSource

/**
 * @see DBRssSourceRepository
 */
interface RssSourceRepository {

    val allSources: Single<List<RssSource>>

    val onlyActive: Single<List<RssSource>>

    fun getById(id: Long): Single<RssSource>

    fun getQueryResult(query: String): Single<FeedlyResponse>

    fun addNewSource(source: RssSource)

    fun updateSource(source: RssSource)

    fun swapSources(from: RssSource, to: RssSource)

    fun deleteSource(source: RssSource)

    fun checkIfExists(source: RssSource): Boolean
    fun deleteSourceByLink(source: RssSource)
}
